### PR TITLE
Adding Tests for RFC1123

### PR DESCRIPTION
**Pull request by Petermarcu**
I'm doing some changes on this in coreclr and noticed we don't have coverage for "r"

### CODE REVIEW

File: src/System.Runtime/tests/System/DateTimeOffsetTests.cs [hunk 1]
```diff
@@ -871,6 +871,16 @@ public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O()
...
 
         [Fact]
+        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_R()
+        {
+            DateTimeOffset expected = DateTimeOffset.MaxValue;
+            string expectedString = expected.ToString("r");
+
+            DateTimeOffset result = DateTimeOffset.ParseExact(expectedString, "r", null, DateTimeStyles.None);
+            Assert.Equal(expectedString, result.ToString("r"));
+        }
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/DateTimeOffsetTests.cs]: This seems to be testing that we can successfully roundtrip via ToString and ParseExact.  Do we also have tests that verify we get the expected string from ToString?  Do we have tests that verify this for a variety of inputs, i.e. various years/months/days/hours/minutes/seconds/offsets/etc.?  If not, you might consider making this a theory that takes both the DateTimeOffset and the expected string, and then having the test do what it's doing now but also compare the generated ToString to the expected one provided. (Same comment/question for the DateTime test later.)

Comment (Petermarcu) [inline src/System.Runtime/tests/System/DateTimeOffsetTests.cs]: Yeah, I was talking to @tarekgh about this a few days ago. The testing here exercises the code paths but doesn't really give great coverage. What I did here is mimic the code path testing we have for the other formats. I plan to file an issue to track taking a more coherent look at our test coverage. I agree that a theory with a variety of inputs would be best. It should be done for all these tests, not just this format. Min value, Max value, combined with 0, min and max offsets. Now, and some interesting hard coded dates that exercise things in the middle. That should be the data set we feed into all the parsing code.

Comment (tarekgh) [inline src/System.Runtime/tests/System/DateTimeOffsetTests.cs]: Yes we should file issue to track add more coverage

**Comment by tarekgh** [comment]
LGTM.

**Comment by Petermarcu** [comment]
I filed the issue to track shoring up these tests with more valid input data.

File: src/System.Runtime/tests/System/DateTimeOffsetTests.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime/tests/System/DateTimeOffsetTests.cs]: Is there a reason we rushed the coreclr change in without having solid tests in place?  I would have much preferred it if we took the extra time to add the tests we wanted for the cases being modified, before merging the changes into coreclr.  It's one thing to have not-great-tests in place for something, but it's another to have not-great-tests in place for something AND make fundamental changes to the implementation.  It should not take very long to add such tests.

**Comment by Petermarcu** [comment]
I'll get some tests added for this case.

**Comment by stephentoub** [comment]
> I'll get some tests added for this case.

Thanks

**Comment by Petermarcu** [comment]
https://github.com/dotnet/corefx/pull/13180

**Pull request merged**